src/PolarCircleResolution.js

```diff
@@ -0,0 +1,119 @@
+import SolarTime from './SolarTime';
+import { dateByAddingDays } from './DateUtils';
+
+export const PolarCircleResolution = {
+  aqrabBalad: 'aqrab-balad',
+  aqrabYaum: 'aqrab-yaum',
+  unresolved: 'unresolved'
+};
+
+const LATITUDE_VARIATION_STEP = 0.5; // Degrees to add/remove at each resolution step
+const UNSAFE_LATITUDE = 65; // Based on https://en.wikipedia.org/wiki/Midnight_sun
+
+const isValidSolarTime = (solarTime) => {
+  if (solarTime) {
+    return solarTime && !isNaN(solarTime.sunrise) && !isNaN(solarTime.sunset);
+  }
+  return false;
+}
+
+const aqrabYaumResolver = (coordinates, date, daysAdded) => {
+  if (daysAdded <= Math.ceil(365 / 2)) {
+    const pastDate = new Date(date.getTime());
+    pastDate.setDate(pastDate.getDate() - daysAdded);
+    const solarTimePast = new SolarTime(pastDate, coordinates);
+    const solarTimePastTomorrow = new SolarTime(dateByAddingDays(pastDate, 1), coordinates);
+
+    if (!isValidSolarTime(solarTimePast) || !isValidSolarTime(solarTimePastTomorrow)) {
```

Comment:
we could simplify having to do both past and future checks by passing in a direction to indicate whether to add or subtract days to do the check. For example 

`const aqrabYaumResolver = (coordinates, date, daysAdded, direction)`

then if the direction is positive we don't increment daysAdded but simply call back with the same daysAdded but with a negative direction. Then on the negative direction pass we do increment daysAdded and then swap back to positive.

Reply:
This is a great enhancement!